scripts/keys.coffee

```diff
@@ -10,95 +10,309 @@
+		robot.brain.set("keyBlue",Key2)
+		Key2	
+	keyYellow = ()->
+		Key1 = robot.brain.get("keyYellow") or ""
+		robot.brain.set("keyYellow",Key3)
+		Key3	
+	keyGreen = ()->
+		Key1 = robot.brain.get("keyGreen") or ""
+		robot.brain.set("keyGreen",Key4)
+		Key4	
+	keyOrange = ()->
+		Key1 = robot.brain.get("keyOrange") or ""
+		robot.brain.set("keyOrange",Key5)
+		Key5	
 
-	
-	robot.respond /i have (a key|the key|key|keys)/i, (msg)->
+    #1this section belongs to "i have keys"
+	robot.respond /i (have the|have) (.+) (key|keys)/i, (msg)->
+		keyname = msg.match[2]
 		name = msg.message.user.name 
 		user = robot.brain.userForName name
-		k = key()
-		if typeof user is 'object'
-			msg.send "Okay #{name} has keys"
-			k = user.name
-		robot.brain.set("key",k)	
-
+		unless keyname not in  ["red", "blue", "yellow", "green", "orange"]
+			if keyname is 'red'
+				key_holder = keyRed()
+				key_holder = user.name
+				robot.brain.set("keyRed",key_holder)
+			else if keyname is 'blue'
+				key_holder = keyBlue()
+				key_holder = user.name
+				robot.brain.set("keyBlue",key_holder)
+			else if keyname is 'Yellow'
+				key_holder = keyYellow()
+				key_holder = user.name
+				robot.brain.set("keyYellow",key_holder)
+			else if keyname is 'Green'
+				key_holder = keyGreen()
+				key_holder = user.name
+				robot.brain.set("keyGreen",key_holder)
+			else if keyname is 'orange'
+				key_holder = keyOrange()
+				key_holder = user.name
+				robot.brain.set("keyOrange",key_holder)
+			if typeof user is 'object'
+				msg.send "Okay #{name} has #{keyname} keys"	
 
-	robot.respond /i (don\'t|dont|do not) (has|have) (the key|key|keys|a key)/i , (msg)->
+	#2this section belongs to "i dont have key"
+	robot.respond /i (do not|don\'t|dont) (has the|have the|has|have) (key|keys)/i , (msg)->
 		name = msg.message.user.name
 		user = robot.brain.userForName name
-		k = key()
-		if k is name
-			k = ""
-		if typeof user is 'object'
-			if k is ""
-				msg.send "Okay #{name} doesn't have keys. Who got the keys then?"
-			else
-				msg.send "Yes , i know buddy, its because #{k} has got the keys"	
-		robot.brain.set("key",k)	
-
+		key_holder1 = keyRed()
+		key_holder2 = keyBlue()
+		key_holder3 = keyYellow()
+		key_holder4 = keyGreen() 
+		key_holder5 = keyOrange()
+		if name is key_holder1
+				key_holder1 = ""	
+				robot.brain.set("keyRed",key_holder1)	
+				msg.send "Okay #{name} doesn't have Red keys. Who got the Red keys then?"
+		else if name is key_holder2
+				key_holder2 = ""	
+				robot.brain.set("keyBlue",key_holder2)	
+				msg.send "Okay #{name} doesn't have Blue keys. Who got the Blue keys then?"
+		else if name is key_holder3
+				key_holder3 = ""	
+				robot.brain.set("keyYellow",key_holder3)	
+				msg.send "Okay #{name} doesn't have Yellow keys. Who got the Yellow keys then?"
+		else if name is key_holder4
+				key_holder4 = ""	
+				robot.brain.set("keyGreen",key_holder4)	
+				msg.send "Okay #{name} doesn't have Green keys. Who got the green keys then?"
```

Comment:
There are a few letter case issues in the code. I am not going to pin point all of them. Go through the code once. `G` should be capital here. It is a good practice to follow the protocols of standard letter casing.